app/main.py

```diff
@@ -1,13 +1,18 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {"line": error["line_number"],
+            "column": error["column_number"],
+            "message": error["text"],
+            "name": error["code"],
+            "source": "flake8"}
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {"errors": [format_linter_error(i) for i in errors],
+            "path": file_path,
+            "status": "failed"} if errors else \
+           {"errors": [], "path": file_path, "status": "passed"}
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [format_single_linter_file(key, value)
+            for key, value in linter_report.items()]
```

Comment:
not sure if it's good practice, I mean yes, sure it's path and errors, but since we work with dict, we prefer, as programmers to see all manipulations with key-value bundle. And I know we should use your pattern for study, but I ask U either U think it's really that better than even k: v, which known by all programmers as representation for key-value. I mean we can use __str__ __repr__ for explanations, but for manipulations aren't k-v preferable?

Reply:
Ofc, as programmers we know what is k, v. But imagine situation with 1000+ lines of code and a loop with `for k, v in..`. Understanding what stands behind "k" and "v" can be really tough in such cases. Why wouldn't we use clear naming if we also know that `path` and `errors` are `key` and `value` respectively?:)

Reply:
Idk I'm sure we can just represent through mentioned above methods for debugging, but in case, where other person checking code, I'd prefer inherited from dict type...
I mean I know proper naming is a great solution in wide cases, just asking personally, so thanks for your pov